Replace ring with aws-lc-rs

maherthomsi opened the pull request:
Issue number:
https://github.com/bottlerocket-os/bottlerocket-update-operator/issues/703

Description of changes:
Replaces dependency on ring to instead use aws-lc-rs for cryptographic operations

Testing done:

Ran `make build` and integration tests to ensure that the replacement did not break anything
Ran `make check-licenses` to validate aws-lc-rs licenses
Terms of contribution:

By submitting this pull request, I agree that this contribution is dual-licensed under the terms of both the Apache License, version 2.0, and the MIT license.

arnaldo2792 commented on Dockerfile:
```diff
@@ -13,6 +13,9 @@ ADD ./ /src/
 RUN cargo fetch --locked --manifest-path /src/Cargo.toml
 
+# Set bindgen clang arguments for cross-compilation targeting Bottlerocket's musl environment
+ENV BINDGEN_EXTRA_CLANG_ARGS="--target=${UNAME_ARCH}-bottlerocket-linux-musl --sysroot=/${UNAME_ARCH}-bottlerocket-linux-musl/sys-root"
```
Before you merge, could you please confirm that the Bottlerocket toolchain was used to compile `aws-lc`?

maherthomsi committed:
build(deps): replace ring with aws-lc-rs

The ring crate is deprecated and unmaintained. AWS-LC-RS provides
equivalent functionality with active support and FIPS compliance.

Signed-off-by: Maher Homsi <maherhom@amazon.com>
